dns: support max_timeout for retry query (#58408)
nodejs/node: doc/api/dns.md

    default timeout.
  * `tries` {integer} The number of tries the resolver will try contacting
    each name server before giving up. **Default:** `4`

>   * `maxTimeout` {integer} The number of max timeout in milliseconds when
>     retry, or `0` to use the default timeout. **Default:** `-1`

Reviewer: ```suggestion
  * `maxTimeout` {integer} The max retry timeout, in milliseconds. Set to `0` to
    use the default timeout. **Default:** `-1`, disabled.
```

Author: Thanks ! The code has been updated.
